bbq/compiler/desugar.go

```diff
@@ -382,6 +587,45 @@ func (d *Desugar) generateConditionsFunction(
 		nil,
 	)
 
+	params := make([]*ast.Parameter, len(parameterList.Parameters))
+	copy(params, parameterList.Parameters)
+
+	// For post conditions, also add `result` as a parameter, if needed.
+	_, needResultVar := d.elaboration.ResultVariableType(functionBlock)
+	if kind == ast.ConditionKindPost && needResultVar {
+		params = append(params, ast.NewParameter(
+			d.memoryGauge,
+			sema.ArgumentLabelNotRequired,
+			ast.NewIdentifier(
+				d.memoryGauge,
+				resultVariableName,
+				pos,
+			),
+
+			ast.NewTypeAnnotation(
+				d.memoryGauge,
+				false,
+
+				// TODO: Pass the proper type here. It can be looked-up from the elaboration (see above).
+				//  However, will have to convert the `sema.Type` to `ast.Type`.
+				//  Currently this `ast.Type` is not needed for the compiler, hence passing a dummy type.
```

Comment:
Here the AST type is not really used by the compiler, so this is probably fine for now. But eventually we would need to pass the proper type, just for consistency.